### DISCUSSION TITLE
Getting the type of the document uploaded

### DISCUSSION THREAD

**Discussion by developerint3030** [pnp/pnpcore#1142] [Q&A] [answered]
Hi,

I have a function that upload different files format into SharePoint: word document, image, text file, etc

I need to return the content type of that file in SharePoint.

The client is asking for instance, if I upload a word document (.docx) I need to return the MIME type:

application/vnd.openxmlformats-officedocument.wordprocessingml.document

Is this something I can get OOB using PnP Core?

Thanks,
Marietta

**Comment by jansenbe** [comment]
There's no native API for that and SharePoint is not storing the mime types, however you can deduct the mime type from the extension. See https://github.com/samuelneff/MimeTypeMap/blob/master/MimeTypeMap.cs for sample code. The same code is also available as nuget package: https://www.nuget.org/packages/MimeTypeMapOfficial

**Comment by developerint3030** [comment, marked as answer]
Thank you!
